dotCMS/build.gradle

```diff
@@ -253,62 +346,319 @@ ext.prepareFelixExternalFolder = { felixExternalFolder ->
+    group = "verification"
+    testClassesDirs = sourceSets.integrationTest.output.classesDirs
+    classpath = sourceSets.integrationTest.runtimeClasspath
+
+
+    // Check if we want to run the tests suite
+    def singleTest = System.getProperty("test.single")
+    if (null != singleTest && singleTest == "com.dotcms.MainSuite"){
+        include "**/MainSuite.class"
+    } else {
+        exclude "**/MainSuite.class"
     }
+
+
+    doFirst
+            {
+                if (project.hasProperty('databaseType')){
+                    systemProperty "databaseType", "$databaseType"
+                }
+            }
+
+
+    // reports.junitXml.destination project.file("$buildDir/test-results/unit-tests/xml")
+    //reports.html.destination project.file("$buildDir/test-results/unit-tests/html")
+    //mustRunAfter(tasks.named('test'))
 }
 
-task copyTestRuntimeLibs(type: Copy) {
-    into "build/libs/test"
-    from configurations.testCompile - configurations.compile
+
+
+tasks.withType(Test) {
+    useJUnit()
+    maxHeapSize = '1G'
+       // jvmArgs = ["--add-opens=java.base/jdk.internal.loader=ALL-UNNAMED", "-javaagent:${configurations.byteBuddyAgent.singleFile}"]
+
+    testLogging {
+        // set options for log level LIFECYCLE
+        events TestLogEvent.FAILED,
+                TestLogEvent.PASSED,
+                TestLogEvent.SKIPPED
+        exceptionFormat TestExceptionFormat.FULL
+
+    }
 }
 
-// Expand the release properties.
-processResources {
-    filesMatching("release.properties") {
-        expand( project.properties )
+
+// End build
+
+//Gradle Custom Task that contains the logic for deploying the app war in the application server
+//It works for dist and development versions
+//Also, copies specific libraries from /libs/buildlibs to tomcat/lib folder
+class DeployWarTomcatTask extends DefaultTask{
+    @Input
+    String tomcatPath
+
+    @TaskAction
+    def deploy(){
+        copyBuildLibs(tomcatPath)
+
+        def zipFile = project.file("$project.buildDir/libs/$project.archivesBaseName" + ".war")
+        project.copy{
+            from project.zipTree(zipFile)
+            into tomcatPath + project.webAppRootFolder.toString()
+        }
+
+        if (project.gradle.startParameter.taskNames.contains("deployWarTomcatTests")) {
+            project.replaceText("$tomcatPath$project.webAppRootFolder/WEB-INF/web.xml", "<!-- TEST FRAMEWORK SERVLETS", "<!-- TEST FRAMEWORK SERVLETS -->")
+            project.replaceText("$tomcatPath$project.webAppRootFolder/WEB-INF/web.xml", "END OF TEST FRAMEWORK SERVLETS -->", "<!-- END OF TEST FRAMEWORK SERVLETS -->")
+        }
     }
+
+    // Copying libs/buildlibs under tomcat8/lib.
+    def copyBuildLibs(buildLibLocation) {
+
+        project.copy {
+            from project.configurations.compileClasspath
+            into buildLibLocation + '/lib'
+            include '**/ojdbc*.jar'
+            include '**/postgresql*.jar'
+            include '**/mysql-connector-java*.jar'
```

Comment:
do we still needing Oracle and mysql?

Reply:
Question for @wezell Do we want to clean these up now or still intending on working with them for the time being?